File: public/static/templates/nodejs/repository/Repository.js
forEach: Aggregate
fileName: {{namePascalCase}}Repository.js
path: {{boundedContext.name}}/repository
---
const db = require('../models')

async function save(body){
    return await db.{{namePascalCase}}.create(body,{
        returning: true
    })
}

async function findall(){
    var results = await db.{{namePascalCase}}.findAll({raw:true})
    console.log('results',results);
    return results
}

async function find_by_id(id){
    var results = await db.{{namePascalCase}}.findOne({
        where:{
            id: id 
        },
        raw:true,
    })
    
    return results
}

async function update(id, body){
    var results = await db.{{namePascalCase}}.update(body,{
        where:{
            id: id,
        },
        raw:true,
        returning:true,
    })
    if (results){
        var result = await find_by_id(id)
        return result
    }
}

async function destroy(id){
    var result = await db.{{namePascalCase}}.destroy({
        where:{
            id : id,
        }
    })
    console.log('result', result)
    return result
}

exports.save = save
exports.read = findall
exports.read_by_id = find_by_id
exports.update = update
exports.delete = destroy

File: public/static/templates/python/repository.py
forEach: Aggregate
fileName: {{namePascalCase}}Repository.py
path: {{boundedContext.name}}
---
from flask import Response, request

import {{namePascalCase}}DB
import util
from {{namePascalCase}} import {{namePascalCase}}

repository = {{namePascalCase}}DB.repository

def Create(request):
    entity = {{namePascalCase}}()
    entity = util.AutoBinding(request.json, entity)
    repository.save(entity)

    return entity
    
def Read(request):
    entity_list = repository.list()
    
    return entity_list

def Read_by_id(id):
    
    ele = repository.find_by_id(id)

    return ele

def Update(id, request):
    
    ele = repository.update(id, request.json)
    
    return ele

def Delete(id):
    ele = repository.delete(id)
    
    return ele